### ISSUE TITLE
GetTypedArrayBytes is wrongly guessing size base on underlying ArrayBuffer

### ISSUE THREAD

**Issue by vorg** [deanm/plask#42]
Consider the following code

```javascript
var image = getImageData();
var textureData1 = new Uint8Array(image.buf, image.offset, image.length);
var textureData2 = new Uint8Array(image.buf.slice(image.offset, image.offset + image.length));
```

```C++
GetTypedArrayBytes(args[6], &textureData1, &size1)
//size1 = 11184952 -> length of the whole image buffer
GetTypedArrayBytes(args[6], &textureData2, &size2)
//size2 = 8388608 -> image.length
```

**Comment by vorg** [comment]
So two Uint8Arrays are seemingly the same but secretly fail to upload the right data.

**Comment by deanm** [comment]
I believe what is wrong is that GetTypedArrayBytes should take in to account ByteOffset and ByteLength in the case that the array is an ArrayBufferView

**Comment by deanm** [comment]
Fixed in aba745fe45346ce1b3b5ce0332e837203ab2394c
